src/main/java/com/example/dog/controller/DogController.java

```diff
@@ -36,8 +37,8 @@ public ResponseEntity<?> updateDog(@PathVariable("id") int id, @Valid @RequestBo
         try {
             Dog addedDog = dao.update(id, dog);
             return new ResponseEntity<>(addedDog, HttpStatus.OK);
-        } catch (IllegalArgumentException e) {
-            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.getMessage());
+        } catch (ObjectNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
```

Comment:
Yep. Now the last change that we need is to introduce a generic Exception Handler. Because in many endpoints we can get ObjectNotFoundException, it makes sense to create a single hander for this this exception and not worry about it anymore. Look for `@ControllerAdvice` [here](https://spring.io/blog/2013/11/01/exception-handling-in-spring-mvc).